crates/proto/src/serialize/binary/encoder.rs

```diff
@@ -439,6 +486,33 @@ impl<'a> BinEncoder<'a> {
     }
 }
 
+/// Determines whether name compression inside RDATA fields.
+#[derive(Clone, Copy)]
+pub enum RdataCompression {
+    /// Compression is allowed inside RDATA.
+    ///
+    /// This is only applicable to well-known record types defined in RFC 1035.
+    CompressionAllowed,
+    /// Compression is not allowed inside RDATA.
+    ///
+    /// This is applicable to any record types defined after RFC 1035. Names must be uncompressed so
+    /// that servers can correctly process messages containing such records even if they don't
+    /// recognize the record type.
+    NoCompression,
+}
+
+/// Determines whether names inside RDATA fields are transformed to lowercase in DNSSEC canonical form.
+#[derive(Clone, Copy)]
+pub enum CanonicalCase {
+    /// Names inside RDATA are transformed to lowercase.
+    ///
+    /// This applies to a list of record types defind in RFC 4034 and modified by RFC 6840, section
```

Comment:
Add links to these RFC sections if possible?

Reply:
I want to keep this as `CanonicalCase` instead of `Case` to emphasize that this only affects encoding when using DNSSEC canonical form, and doesn't affect non-canonical form encoding.

Reply:
It feels like we should collapse to a single mode enum with 3 or 4 modes for name encoding...